plugin/access.lua

```diff
@@ -60,36 +71,57 @@ end
 -- The public entry point to decrypt the BFF cookie and then forward the token to the API
 --
 function _M.run(config)
-ngx.log(ngx.INFO, "Request proxied through BFF plugin")
-    if ngx.req.get_method() == "OPTIONS" then
+
+    local method = ngx.req.get_method() 
+    if method == "OPTIONS" then
         return
     end
 
     local cookie = ck:new()
-    local bff_cookie, err = cookie:get(config.cookie_name_prefix .. "-at")
 
-    if err then
-        ngx.log(ngx.WARN, "Error getting BFF cookie - " .. err)
+    -- First verify the web origin
+    local web_origin = ngx.req.get_headers()["origin"]
+    if not web_origin or web_origin ~= config.trusted_web_origin  then
```

Comment:
Should it be possible to allow a list of trusted web origins? Maybe the API want to allow access from different web clients? Maybe this is a nice-to-have?